### PR TITLE
New cop rule:  `IrreversibleMigration`

### PR DESCRIPTION
**Pull request by AlasdairWallaceMackie**
Should resolve this issue: https://github.com/rubocop/rubocop-sequel/issues/30

According to the [Sequel documentation](https://sequel.jeremyevans.net/rdoc/files/doc/migration_rdoc.html#label-A+Basic+Migration), only certain methods should be placed inside a `change` block in a migration file. Other methods could make the migration irreversible.

This new cop will warn the user under the following conditions inside a `change` block:
- They use a method that's not listed in the documentation
- They use the method `#add_primary_key` with an array argument instead of a symbol

### CODE REVIEW

File: rubocop-sequel.gemspec [hunk 1]
```diff
@@ -17,5 +17,5 @@ Gem::Specification.new do |gem|
 
   gem.required_ruby_version = '>= 2.5'
 
-  gem.add_runtime_dependency 'rubocop', '~> 1.0'
+  gem.add_dependency 'rubocop', '~> 1.0'
```
Comment (AlasdairWallaceMackie) [inline rubocop-sequel.gemspec:20]: Had to make this change due to [failed CI build](https://github.com/rubocop/rubocop-sequel/actions/runs/11420687878/job/31781231537)

image

**Review by cyberdelia — APPROVED** [review]

**Comment by tt** [comment]
This introduced an issue where `change` blocks outside of Sequel migrations are also being detected. See #44 for context.

**Comment by soumyaray** [comment]
The `IrreversibleMigration` rule disallows using `primary_key` and `String` for `create_table` in a `change` block. However, the [Sequel documentation](https://sequel.jeremyevans.net/rdoc/files/doc/migration_rdoc.html#label-A+Basic+Migration) example for how to use `change` in lieu of `up`/`down` blocks uses `primary_key` and `String` methods for a `create_table` in a `change` block. I'm a bit confused here as to why these are not allowed by `IrreversibleMigration`? Would the example documentation also fail this cop?

**Comment by AlasdairWallaceMackie** [comment]
@soumyaray Thanks for bringing this to my attention, the cop should be ignoring all methods within a `create_table` block. I will work on a fix for this

**Comment by soumyaray** [comment]
> @soumyaray Thanks for bringing this to my attention, the cop should be ignoring all methods within a `create_table` block. I will work on a fix for this

Thank you so much for your quick handling of this -- really appreciate the work on this gem.

**Pull request merged**